src/sentry/static/sentry/app/components/group/suggestedOwners.jsx

```diff
@@ -96,151 +112,108 @@ const SuggestedOwners = createReactClass({
-          }}
-          title={ReactDOMServer.renderToStaticMarkup(
-            <div>
-              {author.id ? (
-                <div className="tooltip-owners-name">{author.name}</div>
-              ) : (
-                <div className="tooltip-owners-unknown">
-                  <p className="tooltip-owners-unknown-email">
-                    <span className="icon icon-circle-cross" />
-                    <strong>{author.email}</strong>
-                  </p>
-                  <p>
-                    {t(`Sorry, we don't recognize this member. Make sure to link alternative
-                    emails in Account Settings.`)}
-                  </p>
-                  <hr />
-                </div>
-              )}
-              <ul className="tooltip-owners-commits">
-                {commits.slice(0, 6).map(c => {
-                  return (
-                    <li key={c.id} className="tooltip-owners-commit">
-                      <div style={{whiteSpace: 'pre-line'}}>
-                        {c.message.replace(
-                          /\n\s*\n/g,
-                          '\n'
-                        ) /*repress repeated newlines*/}
-                      </div>
-                      <span className="tooltip-owners-date">
-                        {' '}
-                        - {moment(c.dateCreated).fromNow()}
-                      </span>
-                    </li>
-                  );
-                })}
-              </ul>
-            </div>
-          )}
-        >
-          <Avatar user={author} />
-        </Tooltip>
-      </span>
-    );
-  },
+  /**
+   * Combine the commiter and ownership data into a single array, merging
+   * users who are both owners based on having commits, and owners matching
+   * project ownership rules into one array.
+   *
+   * The return array will include objects of the format:
+   *
+   * {
+   *   actor: <
+   *    SentryTypes.User,  # API expanded user object
+   *    {email, id, name}  # Sentry user which is *not* expanded
+   *    {email, name}      # Unidentified user (from commits)
+   *    {id, name},        # Sentry team (check `type`)
+   *   >,
+   *
+   *   # One or both of commits and rules will be present
+   *
+   *   commits: [...]  # List of commits made by this owner
+   *   rules:   [...]  # Project rules matched for this owner
+   * }
+   */
+  getOwnerList() {
+    const owners = this.state.committers.map(commiter => ({
+      actor: {type: 'user', ...commiter.author},
+      commits: commiter.commits,
+    }));
+
+    this.state.owners.forEach(owner => {
+      const normalizedOwner = {
+        actor: owner,
+        rules: findMatchedRules(this.state.rules || [], owner),
+      };
+
+      const existingIdx = owners.findIndex(o => o.actor.email === owner.email);
+      if (existingIdx > -1) {
+        owners[existingIdx] = {...normalizedOwner, ...owners[existingIdx]};
+      } else {
+        owners.push(normalizedOwner);
+      }
```

Comment:
It would probably make sense to at least make the owners endpoint return something less obtuse than a deeply nested set of lists.

But I didn't want to dig too much into the API for these things.

Reply:
Ok. Maybe that's something we can circle back on? This isn't the first place where we're forced to munge a bunch of data around in the client because the API isn't great.